app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsResponse.kt

```diff
@@ -10,10 +12,30 @@ data class DirectionsResponse(
 
 /**
  * Data model for a single route within the Directions API response. Each route contains an overview
- * polyline, which is used to draw the route on the map.
+ * polyline, which is used to draw the route on the map, and a list of legs, each representing a
+ * segment of the journey between waypoints.
  */
 data class Route(
-    val overviewPolyline: OverviewPolyline // The polyline representing the entire route
+    val overviewPolyline: OverviewPolyline, // The polyline representing the entire route
+    val legs:
+        List<Leg> // List of legs in the route, each leg represents a segment between waypoints
+)
+
+/**
+ * Data model for a single leg within a route. A leg represents the journey between two waypoints,
+ * and includes information such as distance, duration, start and end locations, addresses, and an
+ * encoded polyline.
+ */
+data class Leg(
+    val distanceText: String, // Human-readable representation of the distance (e.g., "3.4 km")
+    val distanceValue: Int, // Distance in meters
+    val durationText: String, // Human-readable representation of the duration (e.g., "15 mins")
+    val durationValue: Int, // Duration in seconds
+    val startAddress: String, // Human-readable address of the start location
+    val endAddress: String, // Human-readable address of the end location
+    val startLocation: LatLng, // LatLng representing the start point of the leg
+    val endLocation: LatLng, // LatLng representing the end point of the leg
+    val overviewPolyline: OverviewPolyline // Encoded polyline representing the path for this leg
```

Comment:
Consider providing default values to avoid null checks or requirements on the data Leg, as done in other classes like TravelContainer.
For example, 
```suggestion
data class Leg(
    val distanceText: String = null, // this is an example
    val distanceValue: Int = 0, // this is an example
    val durationText: String, // Human-readable representation of the duration (e.g., "15 mins")
    val durationValue: Int, // Duration in seconds
    val startAddress: String, // Human-readable address of the start location
    val endAddress: String, // Human-readable address of the end location
    val startLocation: LatLng, // LatLng representing the start point of the leg
    val endLocation: LatLng, // LatLng representing the end point of the leg
    val overviewPolyline: OverviewPolyline // Encoded polyline representing the path for this leg
```
or
```suggestion
data class Leg(
    val distanceText: String, // Human-readable representation of the distance (e.g., "3.4 km")
    val distanceValue: Int, // Distance in meters
    val durationText: String, // Human-readable representation of the duration (e.g., "15 mins")
    val durationValue: Int, // Duration in seconds
    val startAddress: String, // Human-readable address of the start location
    val endAddress: String, // Human-readable address of the end location
    val startLocation: LatLng, // LatLng representing the start point of the leg
    val endLocation: LatLng, // LatLng representing the end point of the leg
    val overviewPolyline: OverviewPolyline // Encoded polyline representing the path for this leg
    ) {
    init {
    require(...) { "Message requirement" }
    }
```

Reply:
He it is a good idea, I will a a block init{}